src/atoms/Input/Input.tsx

```diff
@@ -85,13 +82,14 @@ export class Input extends Component<
 
   public render() {
     const { icon, iconSide } = this.props;
+    const hasIcon = icon ? true : false;
     const formattedIconSide = iconSide && iconSide.toLowerCase();
     const iconElement = icon && (
       <StyledIcon icon={icon} iconSide={formattedIconSide} />
     );
 
     return (
-      <InputContainer>
+      <InputContainer hasIcon={hasIcon}>
```

Comment:
Instead of the `hasIcon` variable, we can use `hasIcon={Boolean(icon)}` to safely convert to a boolean, and TypeScript will still infer the type properly.